src/ImageSharp/Compression/Zlib/Adler32.cs

```diff
@@ -192,35 +199,167 @@ private static unsafe uint CalculateSse(uint adler, ReadOnlySpan<byte> buffer)
                             s2 += s1 += *localBufferPtr++;
                         }
 
-                        if (s1 >= BASE)
+                        if (s1 >= Base)
                         {
-                            s1 -= BASE;
+                            s1 -= Base;
                         }
 
-                        s2 %= BASE;
+                        s2 %= Base;
                     }
 
                     return s1 | (s2 << 16);
                 }
             }
         }
+
+#if NET5_0_OR_GREATER
+
+        // Based on: https://github.com/chromium/chromium/blob/master/third_party/zlib/adler32_simd.c
+        [MethodImpl(InliningOptions.HotPath | InliningOptions.ShortMethod)]
+        private static unsafe uint CalculateArm(uint crc, ReadOnlySpan<byte> buffer)
+        {
+            // Split Adler-32 into component sums.
+            uint s1 = crc & 0xFFFF;
+            uint s2 = (crc >> 16) & 0xFFFF;
+            int len = buffer.Length;
+            int bufferOffset = 0;
+
+            // Process the data in blocks.
+            long blocks = len / BlockSize;
+            len -= (int)(blocks * BlockSize);
+            fixed (byte* bufferPtr = buffer)
+            {
+                while (blocks != 0)
+                {
+                    uint n = Nmax / BlockSize;
+                    if (n > blocks)
+                    {
+                        n = (uint)blocks;
+                    }
+
+                    blocks -= n;
+
+                    // Process n blocks of data. At most nMax data bytes can be
+                    // processed before s2 must be reduced modulo Base.
+                    var vs2 = Vector128.Create(0, 0, 0, s1 * n);
+                    Vector128<uint> vs1 = Vector128<uint>.Zero;
```

Comment:
```suggestion
                    Vector128<uint> vs1 = Vector128<uint>.Zero;
                    Vector128<uint> vs2 = vs1.WithElement(3, s1 * n);
```
I don't know if codegen for ARM will be better this way...don't have a test-setup for this now.
But for [amd64](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAJI8YUCAAcAymIBuvMDFwBuGvSatOPfoJEYovLrgUqaagMwNuubADMYTUgwDCDAN40GXhhIOzsGA7MSAwAFABqMGAY0OSkABwAPByGGAB8aAyR0bEJyalpAJQMAGIQEKEpPAy45AByhZ7eHtTebVlRMVBxSVXpDLK4TgC8HTndCSzOsAEwoXSZCwxLtQ2qre1e2V09+TxpA7UMo9u5vQUsAFpiEOubXk2bxADsYYPkmYOkhXfeAL6PBiA3y8fyBDQhCKdM57dKZU4Tc77YoAIWwUEqqRq9UaGy8LXuYx2eT6B3exyJMNJVxuv02CN2pMOI0OrAA6rwMAALACiABsYAIeKFzJlVj9AW1Jd4Xm9ap8hhK8QwAdQ/kA===) codegen is better, and I assume the JIT will emit similar code for ARM.